src/software/new_geom/polynomial.h

```diff
@@ -0,0 +1,162 @@
+#pragma once
+
+#include <cmath>
+#include <vector>
+
+#include "software/geom/point.h"
+
+/**
+ * Polynomial is a representation of a polynomial
+ * that can calculate values at a given input
+ */
+class Polynomial
+{
+   public:
+    // Threshold for coefficient comparison.
+    // http://www.cplusplus.com/forum/beginner/95128/
+    static constexpr double EPSILON = 1e-15;
```

Comment:
Will be changed once #970 is merged, where `EPSILON` is defined in `GeomConstants`